### PR TITLE
[tracker] get some flash space back

### PR DESCRIPTION
**Pull request by avtolstoy**
### Description

Gets rid of a couple of things on Tracker to get some flash space back:
1. CC310 CryptoCell SHA512
2. Full LwIP IPv6 support (partial is left so sockaddr6_in and the rest are still available)
3. system_error messages

There are still some other things to throw away but for now this should be good enough for #2660 and #2675.

```
   text	   data	    bss	    dec	    hex	filename
 513960	   3038	  60814	 577812	  8d114	../../../build/target/system-part1/platform-26-m/system-part1.elf
```

So, we are at around 23.5K after the changes.

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

**Review by sergeuz — APPROVED** [review]
Awesome, thanks!

File: hal/src/nRF52840/mbedtls/sha512_disable.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "crys_hash.h"
+
+uint32_t __wrap_InitSwHash512(void* pCtx) {
```
Comment (eberseth) [inline hal/src/nRF52840/mbedtls/sha512_disable.c:20]: Would we be able to do this with SHA1 as well or do we still use it somewhere?

Comment (avtolstoy) [reply hal/src/nRF52840/mbedtls/sha512_disable.c:20]: We do not need anything other than SHA256 (for cloud and BLE).

**Comment by eberseth** [comment]
Commit beb5c24c7 is not running stably on my Monitor One.  It keeps failing on connect.
```
0000003391 [system.nm] INFO: State changed: IFACE_LINK_UP -> IP_CONFIGURED
0000003392 [net.ppp.client] TRACE: PPP phase -> Running
0000003392 [net.ppp.client] TRACE: PPP thread event UP data=0
0000003393 [net.ppp.client] TRACE: State CONNECTING -> CONNECTED
0000003395 [net.pppncp] TRACE: Negotiated MTU: 1500
0000003396 [system] INFO: Cloud: connecting
0000003400 [lwip] TRACE: lwip_connect: invalid address
0000003400 [system] ERROR: Cloud socket=0, failed to connect to 3.222.143.118#5684, errno=5
0000003402 [system] WARN: Cloud socket connection failed: -1
0000003723 [lwip] TRACE: lwip_connect: invalid address
0000003725 [system] WARN: Internet test failed: network
0000003725 [system] WARN: Handling cloud error: 2
0000004425 [system] INFO: Cloud: connecting
0000004427 [lwip] TRACE: lwip_connect: invalid address
0000004429 [system] ERROR: Cloud socket=0, failed to connect to 3.222.143.118#5684, errno=5
0000004429 [system] WARN: Cloud socket connection failed: -1
0000004431 [lwip] TRACE: lwip_connect: invalid address
0000004431 [system] WARN: Internet test failed: network
0000004433 [system] WARN: Handling cloud error: 2
```

**Comment by avtolstoy** [comment]
@eberseth Minor issues with LwIP, should work fine as of latest commit.

**Comment by eberseth** [comment]
> @eberseth Minor issues with LwIP, should work fine as of latest commit.

Same error as before.  These are the commits I built against:
```
$ git last
commit 7d2e28c0e5c03754e580b8c08e94a436dc33d576 (HEAD -> feature/tracker-flash-savings, origin/feature/tracker-flash-savings)
Author: Andrey Tolstoy <avtolstoy@gmail.com>
Date:   Wed Aug 23 22:40:14 2023 +0700

    [third_party] lwip: update submodule reference

$ pushd third_party/lwip/lwip

$ git last
commit af3134ca49c7c37289723a02454e47b6a2a08cf9 (HEAD, origin/particle)
Author: Andrey Tolstoy <avtolstoy@gmail.com>
Date:   Fri Aug 18 01:03:06 2023 +0700

    LWIP_IPV6_DEFINES_ONLY
```

**Comment by avtolstoy** [comment]
@eberseth Please update submodules. https://github.com/particle-iot/lwip/commit/1001b17868a17cf0d594e069c172794cf0c1416a is the required commit for `third_party/lwip`

**Comment by eberseth** [comment]
> @eberseth Please update submodules. [particle-iot/lwip@1001b17](https://github.com/particle-iot/lwip/commit/1001b17868a17cf0d594e069c172794cf0c1416a) is the required commit for `third_party/lwip`

Got everything sorted out and is working as expected now.  Thanks!

**Review by eberseth — APPROVED** [review]

**Comment by avtolstoy** [comment]
All tests passed as well:
```
  588 passing (2h)
  28 pending
```

**Pull request merged**